bundles/org.openhab.binding.vallox/src/main/java/org/openhab/binding/vallox/internal/se/telegram/Telegram.java

```diff
@@ -0,0 +1,266 @@
+        this.state = state;
+        this.bytes = telegram;
+    }
+
+    /**
+     * Get sender of this telegram
+     *
+     * @return sender
+     */
+    public Byte getSender() {
+        return bytes[1];
+    }
+
+    /**
+     * Get receiver of this telegram
+     *
+     * @return receiver
+     */
+    public Byte getReceiver() {
+        return bytes[2];
+    }
+
+    /**
+     * Get variable of this telegram
+     *
+     * @return variable
+     */
+    public Byte getVariable() {
+        return bytes[3];
+    }
+
+    /**
+     * Get value of this telegram
+     *
+     * @return value
+     */
+    public Byte getValue() {
+        return bytes[4];
+    }
+
+    /**
+     * Get checksum of this telegram
+     *
+     * @return checksum
+     */
+    public Byte getCheksum() {
+        return bytes[5];
+    }
+
+    /**
+     * Get state details as string
+     */
+    public String stateDetails() {
+        return state.stateDetails;
+    }
+
+    /**
+     * Get telegram as string.
+     */
+    @Override
+    public String toString() {
+        return HexUtils.bytesToHex(bytes, "-");
+    }
+
+    /**
+     * Process telegram and return a map of channels to update
+     */
+    public Map<String, State> parse(String channelID, ValloxExpiringCacheMap cache) {
+        channelsToUpdate.clear();
+        ValloxChannel valloxChannel = ChannelMapper.getValloxChannel(channelID);
+        if (valloxChannel instanceof MultipleValueChannel) {
+            Collection<String> subchannels = valloxChannel.getSubChannels();
+            for (String channel : subchannels) {
+                ValloxChannel vc = ChannelMapper.getValloxChannel(channel);
+                State state = vc.convertToState(getValue());
+                channelsToUpdate.put(channel, state);
+            }
+        } else if (valloxChannel instanceof TemperatureChannel) {
+            State state = valloxChannel.convertToState(getValue());
+            channelsToUpdate.put(channelID, state);
+            calculateEfficiencies(cache);
+        } else if (channelID.contains("status#co2")) {
+            calculateCO2(cache);
+        } else if (channelID.contains("setting#co2SetPoint")) {
+            calculateCO2SetPoint(cache);
+        } else {
+            State state = valloxChannel.convertToState(getValue());
+            channelsToUpdate.put(channelID, state);
+        }
+        logger.debug("Channels parsed from telegram {}", channelsToUpdate);
+        return channelsToUpdate;
+    }
+
+    /**
+     * Calculate efficiencies from measured temperatures. This is specified by Vallox.
+     *
+     * @param cache the cache where temperatures are fetched
+     */
+    public void calculateEfficiencies(ValloxExpiringCacheMap cache) {
+        try {
+            if (!cache.containsTemperatures()) {
+                return;
+            }
+            int tempInside = ValloxSEConstants.TEMPERATURE_MAPPING[Byte.toUnsignedInt(cache.getValue((byte) 0x34))];
```

Comment:
You should probably retrieve from the cache using constants.

Reply:
Probably?

Reply:
I changed the ChannelMapper to enum as suggested and it solves this. Constants are now used more widely.